server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByWithoutValueFilterDataSet.java

```diff
@@ -114,159 +97,194 @@ protected RowRecord nextWithoutConstraint() throws IOException {
     }
     hasCachedTimeInterval = false;
     RowRecord record = new RowRecord(curStartTime);
-    AggregateResult[] aggregateResultList = new AggregateResult[paths.size()];
-    for (Map.Entry<Path, List<Integer>> entry : pathToAggrIndexesMap.entrySet()) {
-      List<AggregateResult> aggregateResults;
-      try {
-        aggregateResults = nextIntervalAggregation(entry);
-      } catch (QueryProcessException e) {
-        throw new IOException(e);
-      }
-      int index = 0;
-      for (int i : entry.getValue()) {
-        aggregateResultList[i] = aggregateResults.get(index);
-        index++;
+    timeRange = new TimeRange(curStartTime, curEndTime - 1);
+
+    AggregateResult[] fields = new AggregateResult[paths.size()];
+
+    try {
+      for (Entry<Path, GroupByExecutor> pathGroupByExecutorEntry : pathExecutors.entrySet()) {
+        GroupByExecutor executor = pathGroupByExecutorEntry.getValue();
+        executor.resetAggregateResults();
+        List<Pair<AggregateResult, Integer>> aggregations = executor.calcResult();
+        for (int i = 0; i < aggregations.size(); i++) {
+          fields[aggregations.get(i).right] = aggregations.get(i).left;
+        }
       }
+    } catch (QueryProcessException e) {
+      logger.error("GroupByWithoutValueFilterDataSet execute has error,{}", e);
```

Comment:
no need to use {},just
logger.error("GroupByWithoutValueFilterDataSet execute has error ", e);

Reply:
fixed